plugin/storage/factory_test.go

```diff
@@ -129,11 +132,34 @@ func TestCreate(t *testing.T) {
 	assert.EqualError(t, err, "Archive storage not supported")
 
 	mock.On("CreateSpanWriter").Return(spanWriter, nil)
+	m := metrics.NullFactory
+	l := zap.NewNop()
+	mock.On("Initialize", m, l).Return(nil)
+	f.Initialize(m, l)
 	w, err = f.CreateSpanWriter()
 	assert.NoError(t, err)
 	assert.Equal(t, spanWriter, w)
 }
 
+func TestCreateDownsamplingWriter(t *testing.T) {
+	f, err := NewFactory(defaultCfg())
+	assert.NoError(t, err)
+	assert.NotEmpty(t, f.factories[cassandraStorageType])
+	mock := new(mocks.Factory)
+	f.factories[cassandraStorageType] = mock
+	spanWriter := new(spanStoreMocks.Writer)
+	mock.On("CreateSpanWriter").Return(spanWriter, nil)
+
+	f.DownsamplingRatio = 0.5
+	m := metrics.NullFactory
+	l := zap.NewNop()
+	mock.On("Initialize", m, l).Return(nil)
+
+	f.Initialize(m, l)
+	_, err = f.CreateSpanWriter()
```

Comment:
Don't you want to check that the returned writer is the DownsamplingWriter?

Reply:
Also, it would make sense to test that when DownsamplingRatio == 1 then the mock `spanWriter` is returned (i.e. data driven test with 2 param values).